src/QsCompiler/CompilationManager/ContextBuilder.cs

```diff
@@ -307,17 +307,17 @@ internal static int FindByValue(this IReadOnlyList<CodeFragment> list, CodeFragm
         /// </summary>
         internal static CodeFragment.TokenIndex? GetNonEmptyParent(this CodeFragment.TokenIndex tIndex)
```

Comment:
I may be missing something here, since I think you should run into issues; All of the methods edited in this file now actually modify the given token index. I think it is confusing, especially since the methods return a token (I would hence expect them to not modify the given one). What am I missing?

Reply:
Can you clarify why you think these methods modify the token index? `Previous()` and `Next()` return a new copy of the token index. They do not mutate it.

Reply:
You are right.